docs/releases/upgrading/upgrade-v12.0.md

```diff
@@ -0,0 +1,204 @@
+    invenio-cli, version 1.3.1
+    ```
+
+!!! info "Virtual environments"
+
+    In case you are not inside a virtual environment, make sure that you prefix each `invenio`
+    command with `pipenv run`.
+
+
+**Local development**
+
+Changing the Python version in your development environment highly
+depends on your setup, and there is no golden rule.
+One example would be to use [PyEnv](https://github.com/pyenv/pyenv).
+
+You should delete your virtualenv before running `invenio-cli` or `pipenv` commands below.
+
+!!! warning "Risk of losing data"
+
+    Your virtual env folder contains uploaded files in InvenioRDM, in `var/instance/data`.
+    If you need to keep such files, make sure you copy them over to the new virtual env in the same location.
+
+### Upgrade InvenioRDM
+
+Make sure that your virtual env is now running with at least Python 3.9. (InvenioRDM v12
+is also tested with Python 3.12.)
+
+There are two options to upgrade your system.
+
+#### Option 1
+
+This approach upgrades the dependencies in place. Your virtual environment for the
+v11 version will be gone afterwards.
+
+```bash
+cd <my-site>
+
+# Upgrade to InvenioRDM v12
+invenio-cli packages update 12.0.0
+pipenv uninstall flask-babelex
+
+# Re-build assets
+invenio-cli assets build
+```
+
+#### Option 2
+
+This approach will create a new virtual environment and leaves the v11 as it is.
+If you are using a docker image on your production instance this will be the
+option you chose.
+
+##### Step 1
+- create a new virtual environment
+- activate your new virtual environment
+- install `invenio-cli` by `pip install invenio-cli`
+
+##### Step 2
+Update the file `<my-site>/Pipfile`.
+
+```diff
+[packages]
+---invenio-app-rdm = {extras = [...], version = "~=11.0.0"}
++++invenio-app-rdm = {extras = [...], version = "~=12.0.0"}
+```
+
+Due to a dependency upgrade, update the following line.
+```diff
+---my-site = {editable="True", path="./site"}
++++my-site = {editable=true, path="./site"}
+```
+
+##### Step 3
+Update the `Pipfile.lock` file
+
+```bash
+invenio-cli packages lock
+```
+
+##### Step 4
+install InvenioRDM v12
+
+```bash
+invenio-cli install
+```
+
+
+### Database migration
+
+Execute the database migration:
+
+```bash
+# Execute the database migration
+invenio alembic upgrade
+```
+
+### Declare usage statistics processing queues
+
+```shell
+invenio queues declare
+```
+
+### Data migration
+
+Execute the data migration
+
+```bash
+pipenv run invenio shell $(find $(pipenv --venv)/lib/*/site-packages/invenio_app_rdm -name migrate_11_0_to_12_0.py)
+```
+
+### Reindex
+
+```bash
+invenio index destroy --yes-i-know
+invenio index init
+invenio rdm rebuild-all-indices
+```
+
+In v12 the statistics will be stored in the index. The statistic indices are
+created by templates and therefore not affected by `invenio index destroy
+--yes-i-know`. The statistic indices will remain in the list of indices and are
+totally functional also after a reinit.
```

Comment:
```suggestion
In v12, record statistics will be stored in search indices rather than the database.
These indices are created based on *index templates* which are not affected by affected by `invenio index destroy --yes-i-know`.
So while this command will wipe the data and reset the numbers, the feature will stay functional after a re-init.
```
A bit of clarification from my current understanding, but I haven't tested this right now...

Reply:
i think your suggestion could cause some understanding problems `so while this command will wipe the data and reset the number` could be interpreted in a way that also the statistic indices will be removed. this is not the case. the statistic indices will stay also if i do a `invenio index destroy --yes-i-know`. tried in test and production.

Reply:
Ah, I see, thanks for the clarification!
I've updated the suggestion, is it more accurate now?